jsonrpc/examples/server_custom_codec.rs

```diff
@@ -42,14 +42,14 @@ impl Encoder for CustomJsonCodec {
     fn encode(
         &self,
         src: &Self::EnMessage,
-        dst: &mut [u8],
+        dst: &mut ByteBuffer,
     ) -> std::result::Result<usize, Self::EnError> {
         let msg = match serde_json::to_string(src) {
             Ok(m) => m,
             Err(err) => return Err(Error::Encode(err.to_string())),
         };
         let buf = msg.as_bytes();
-        dst[..buf.len()].copy_from_slice(buf);
+        dst.extend_from_slice(buf);
```

Comment:
At this stage we don't need the source buf anymore (we can save the length if needed). I see extend_from_slice is internally calling copy_from_slice, and in this case it could be moved instead to save a copy (using Vec::append()).